frontend/src/metabase-types/api/dataset.ts

```diff
@@ -12,8 +12,9 @@ export type RowValues = RowValue[];
 export interface DatasetColumn {
   id?: FieldId;
   name: string;
-  display_name: string;
-  source: string;
+  display_name?: string;
```

Comment:
Added these new and optional fields based on the API call response from `/api/card/{cardId}/query`, where the `data > cols[0]` looks something like:
```
{
    "description": "blah",
    "semantic_type": "type/PK",
    "table_id": 2,
    "coercion_strategy": null,
    "name": "ID",
    "settings": null,
    "source": "fields",
    "field_ref": [ "field", 37, null ],
    "effective_type": "type/BigInteger",
    "nfc_path": null,
    "parent_id": null,
    "id": 37,
    "visibility_type": "normal",
    "display_name": "ID",
    "fingerprint": null,
    "base_type": "type/BigInteger"
}
```

Reply:
Could you make `display_name` and `description` required, like this: `string | null` and set nulls in `createMockColumn` for missing fields? 

It actually cannot be undefined, and it's important to have correct types. For instance, 
```
const { description = "" } = column;
description.length;  
```
would throw because default values are specified only for `undefined` and not `null`.

Reply:
Ah gotcha, I learned something today 😄 . Would this be an issue for any of the other keys? Or just these two